src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -667,11 +618,11 @@ MulticopterPositionControl::Run()
 			// will remain NAN. Given that the PositionController cannot generate a position-setpoint, this type of setpoint is always equal to the input to the
 			// PositionController.
 			vehicle_local_position_setpoint_s local_pos_sp{};
-			local_pos_sp.timestamp = time_stamp_now;
 			_control.getLocalPositionSetpoint(local_pos_sp);
 
 			// Publish local position setpoint
 			// This message will be used by other modules (such as Landdetector) to determine vehicle intention.
+			local_pos_sp.timestamp = hrt_absolute_time();
```

Comment:
on previous behavior all the timestamp are the same, but now there might be slightly different because not using time_stamp_now.
is that ok?

Reply:
The timestamp field should be treated as uORB metadata (publication time). Propagating a previous timestamp skews the logs and hides latency.